### PR TITLE
Add Trino grammar

### PR DESCRIPTION
**Pull request by etckun**
[Trino](https://github.com/trinodb/trino) is a popular SQL dialect widely used in data science / data analytics field. I would like contribute this language into the repository. Thanks.

### CODE REVIEW

**Comment by kaby76** [comment]
@KvanTTT We have a "symbol conflict" not considered before: "base" as in ["base=" grammar labeling](https://github.com/etckun/grammars-v4/blob/2bc078a4c657f61039c4bb30fe4e942cf53050df/sql/trino/TrinoParser.g4#L520) causes compilation errors in the CSharp target. Change to "base_".

@etckun Is there a reason why you [pass an inherited attribute](https://github.com/etckun/grammars-v4/blob/2bc078a4c657f61039c4bb30fe4e942cf53050df/sql/trino/TrinoParser.g4#L467),
but never use "value"? Remove, and the grammar becomes target-independent.

With these two changes, the grammar works for CSharp, Go, Dart, Java (of course), and maybe the rest.

**Comment by kaby76** [comment]
Looks good!

File: sql/trino/TrinoParser.g4 [hunk 1]
```diff
@@ -0,0 +1,867 @@
...
+    | OMIT_ EMPTY_ MATCHES_
+    | WITH_ UNMATCHED_ ROWS_
+    ;
+
+skipTo
+    : SKIP_ TO_ NEXT_ ROW_
+    | SKIP_ PAST_ LAST_ ROW_
+    | SKIP_ TO_ FIRST_ identifier
+    | SKIP_ TO_ LAST_ identifier
+    | SKIP_ TO_ identifier
```
Comment (KvanTTT) [inline sql/trino/TrinoParser.g4:393]:
```suggestion
    : SKIP_ 
           ( TO_ (NEXT_ ROW_ | (FIRST_ | LAST_)? identifier)
           | PAST_ LAST_ ROW_
           )
```

File: sql/trino/TrinoParser.g4 [hunk 2]
```diff
@@ -0,0 +1,867 @@
...
+jsonQueryWrapperBehavior
+    : WITHOUT_ ARRAY_?
+    | WITH_ (CONDITIONAL_ | UNCONDITIONAL_)? ARRAY_?
+    ;
+
+jsonQueryBehavior
+    : ERROR_
+    | NULL_
+    | EMPTY_ ARRAY_
+    | EMPTY_ OBJECT_
```
Comment (KvanTTT) [inline sql/trino/TrinoParser.g4:609]:
```suggestion
    | EMPTY_ (ARRAY_ | OBJECT_)
```

File: sql/trino/TrinoParser.g4 [hunk 3]
```diff
@@ -0,0 +1,867 @@
...
+    | DOUBLE_ PRECISION_                                                             #doublePrecisionType
+    | ARRAY_ LT_ type GT_                                                           #legacyArrayType
+    | MAP_ LT_ keyType=type COMMA_ valueType=type GT_                                  #legacyMapType
+    | type ARRAY_ (LSQUARE_ INTEGER_VALUE_ RSQUARE_)?                                          #arrayType
+    | identifier (LPAREN_ typeParameter (COMMA_ typeParameter)* RPAREN_)?                     #genericType
+    ;
+
+rowField
+    : type
+    | identifier type;
```
Comment (KvanTTT) [inline sql/trino/TrinoParser.g4:678]:
```suggestion
    : type
    | identifier type
    ;
```

**Review by KvanTTT — COMMENTED** [review]
Thanks for the contribution, but there are some minor issues to resolve.

**Comment by KvanTTT** [comment]
> @KvanTTT We have a "symbol conflict" not considered before: "base" as in ["base=" grammar labeling](https://github.com/etckun/grammars-v4/blob/2bc078a4c657f61039c4bb30fe4e942cf53050df/sql/trino/TrinoParser.g4#L520) causes compilation errors in the CSharp target. If "base" is changed to "base_" in TrinoParser.g4, the grammar works for CSharp.

Ideally, it would be great to create issues for such cases, or append to an existing one.

**Comment by etckun** [comment]
Thank you for the suggestions @kaby76  @KvanTTT. I have updated the files accordingly :)

**Pull request merged**